packages/xo-web/src/common/editable/index.js

```diff
@@ -341,74 +332,92 @@ export class Number extends Component {
   }
 }
 
-export class Select extends Editable {
+class SimpleSelect_ extends Editable {
   static propTypes = {
-    options: PropTypes.oneOfType([PropTypes.array, PropTypes.object])
-      .isRequired,
-    renderer: PropTypes.func,
-  }
-
-  componentWillReceiveProps(props) {
-    if (
-      props.value !== this.props.value ||
-      props.options !== this.props.options
-    ) {
-      this.setState({
-        valueKey: findKey(props.options, option => option === props.value),
-      })
-    }
+    optionRenderer: PropTypes.func,
+    value: PropTypes.oneOfType([PropTypes.oneOf([null]), PropTypes.object]),
   }
 
   get value() {
-    return this.props.options[this.state.valueKey]
-  }
-
-  _onChange = event => {
-    this.setState({ valueKey: getEventValue(event) }, this._save)
+    return this.state.value
   }
 
-  _optionToJsx = (option, key) => {
-    const { renderer } = this.props
+  _onChange = value => this.setState({ value }, this._save)
 
+  _renderDisplay() {
+    const { children, optionRenderer, value } = this.props
     return (
-      <option key={key} value={key}>
-        {renderer ? renderer(option) : option}
-      </option>
+      children || (
+        <span>
+          {optionRenderer !== undefined
+            ? optionRenderer(value)
+            : value != null
+            ? value.label
+            : _('noValue')}
+        </span>
+      )
     )
   }
 
-  _onEditionMount = ref => {
-    // Seems to work in Google Chrome (not in Firefox)
-    ref && ref.dispatchEvent(new window.MouseEvent('mousedown'))
-  }
-
-  _renderDisplay() {
-    const { children, renderer, value } = this.props
+  _renderEdition = () => (
+    <FormSelect
+      {...this.props}
+      autoFocus
+      onBlur={this._closeEdition}
+      onChange={this._onChange}
+      onKeyDown={this._onKeyDown}
+      openOnFocus
+    />
+  )
+}
 
-    return children || <span>{renderer ? renderer(value) : value}</span>
+class MultiSelect_ extends Editable {
+  static propTypes = {
+    optionRenderer: PropTypes.func,
+    value: PropTypes.oneOfType([PropTypes.oneOf([null]), PropTypes.array]),
   }
 
-  _renderEdition() {
-    const { saving, valueKey } = this.state
-    const { options } = this.props
+  get value() {
+    return this.state.nextValue === undefined
+      ? this.props.value
+      : this.state.nextValue
+  }
 
+  _renderDisplay() {
+    const { children, optionRenderer, value } = this.props
     return (
-      <select
-        autoFocus
-        className={classNames('form-control', styles.select)}
-        onBlur={this._closeEdition}
-        onChange={this._onChange}
-        onKeyDown={this._onKeyDown}
-        readOnly={saving}
-        ref={this._onEditionMount}
-        value={valueKey}
-      >
-        {map(options, this._optionToJsx)}
-      </select>
+      children || (
+        <span>
+          {optionRenderer !== undefined
+            ? optionRenderer(value)
```

Comment:
So when we use `optionRenderer`, we need to handle both an option (object) and a value (array in multi)?
Let's do something like `map(value, optionRenderer || 'label').join(', ')` here and if we want something fancier for the value rendering, we can use the children.